docs/enhancements/clusterpool-inventory.md

```diff
@@ -0,0 +1,186 @@
+      - [Adding An Inventory](#adding-an-inventory)
+      - [Adding An Entry to the Inventory](#adding-an-entry-to-the-inventory)
+      - [Removing An Entry from the Inventory](#removing-an-entry-from-the-inventory)
+      - [Deleting The Inventory](#deleting-the-inventory)
+    - [Maintaining the lease of the ClusterDeploymentCustomization](#maintaining-the-lease-of-the-clusterdeploymentcustomization)
+    - [Fairness](#fairness)
+        
+## Summary
+
+As a cluster administrator, I want to be able to use ClusterPools to provision clusters on on-prem clouds like vSphere, OpenStack and Baremetals.
+
+## Problem Statement
+Provisioning a cluster requires configuration unique to that cluster.
+For example, the cluster's name is used to build the hostnames for the API and console.
+_Someone_ has to handle the DNS that resolves those hostnames to the IPs of the services created by the installer.
+One option is to create your ClusterDeployment with `ManageDNS=True` and we'll create a DNSZone on the fly.
+
+But that doesn't work for some cloud providers such a vSphere, OpenStack and Baremetals.
+
+In such cases, one solution is to configure DNS manually and then create your ClusterDeployment with the right `Name` and `BaseDomain` so that the assembled hostnames match those entries. This is fine if you're creating ClusterDeployments by hand, but breaks down if ClusterDeployment names are being generated with random slugs, as is the case with ClusterPools.
+
+## Proposal
+### Summary
+Allow ClusterPool to accept an inventory of ClusterDeploymentCustomizations that will have install config json patches to be used when generating ClusterDeployments for the pool.
+
+### `ClusterPool.Spec.Inventory`
+Add a field to `ClusterPool.Spec` called `Inventory`.
+
+It has an array of object references to ClusterDeploymentCustomization custom resource. This CR will have a json patch (RFC 6902) for default install config generated by clusterpool controller. The json patch will have cluster specific changes, for vSphere this corresponds to configured DNS hostnames
+
+```yaml
+spec:
+  inventory:
+    ClusterDeploymentCustomizations:
+      - name: foo-cluster-deployment-customization
+      - name: bar-cluster-deployment-customization
+      - name: baz-cluster-deployment-customization
+```
+
+and ClusterDeploymentCustomization CR will look like
+```yaml
+apiVersion: v1
+kind: ClusterDeploymentCustomization
+metadata:
+  name: foo-cluster-deployment-customization
+  namespace: my-project
+spec:
+  installConfigPatches:
+    - op: replace
+      path: metadata/name
+      value: foo
+status:
+  clusterDeploymentRef: 
+    name: foo
+    namespace: foo-namespace
+  conditions:
+    - lastProbeTime: "2020-11-05T14:49:26Z"
+      lastTransitionTime: "2020-11-05T14:49:26Z"
+      message: Currently in use by cluster deployment foo of clusterpool foo-pool
+      reason: ClusterDeploymentCustomizationInUse
+      status: "false"
```

Comment:
I don't know how it was addressed in other cases, but assuming that a ClusterDeploymentCustomization is an object that will exist for a long time, might be used by number of inventories and will be checked regularly for availability, wouldn't the number of such log entries grow repidily and might become unusable?
Maybe keep such log entries in other resources such as CloudDeployment instance for availability check,  and validation in ClusterPool?